doc/reference/reference_sql/sql_statements_and_clauses.rst

```diff
@@ -346,33 +345,37 @@ Two column values in a SCALAR column can have two different primitive data types
       SELECT * FROM t WHERE s1 > 'a';
 
    The comparison is valid, because Tarantool knows the ordering of X'41' and 'a'
-   in Tarantool/NoSQL 'scalar'.
+   in Tarantool/NoSQL 'scalar' -- this is a case where the primitive type matters.
 #. The result data type of :ref:`min/max <sql_aggregate>` operation on a column defined as SCALAR
-   is the data type of the minimum/maximum operand, unless the result value
-   is NULL. For example:
+   is SCALAR.
+   Users will need to know the underlying primitive type of the result in advance. For example:
 
    .. code-block:: sql
 
       CREATE TABLE t (s1 INTEGER, s2 SCALAR PRIMARY KEY);
       INSERT INTO t VALUES (1, X'44'), (2, 11), (3, 1E4), (4, 'a');
-      SELECT min(s2), hex(max(s2)) FROM t;
+      SELECT cast(min(s2) AS INTEGER), hex(cast(max(s2) as VARBINARY)) FROM t;
 
    The result is: ``- - [11, '44',]``
 
    That is only possible with Tarantool/NoSQL scalar rules, but ``SELECT SUM(s2)``
    would not be legal because addition would in this case require implicit casting
-   from VARBINARY to a number, which is not sensible.
-#. The result data type of a primitive combination is never SCALAR because we
-   in effect use TYPEOF(item) not the defined data type.
+   from VARBINARY to a numeric, which is not sensible.
+#. The result data type of a primitive combination is sometimes SCALAR although we
+   in effect use the primitive data type not the defined data type.
    (Here we use the word "combination" in the way that the standard document
```

Comment:
Hello @pgulutzan,
Is [this](https://www.sis.se/api/document/preview/916885/) the standard document you are talking about?

Reply:
Not exactly, that's the technical corrigendum i.e. the list of corrections for SQL:2011. We care about SQL:2016 as mentioned earlier in the manual, and we care about the actual document not a corrigendum. However, there wasn't significant change between 2011 and 2016 for this or indeed for most of the features that Tarantool tries to support, so we can just say "standard document".